src/reducers/actionHelpers/datasource.js

```diff
@@ -96,16 +95,24 @@ export const dismissEditor = (state, { stateKey }) => {
 };
 
 export const removeRow = (state, { stateKey, rowIndex }) => {
+    const existingState = state.get(stateKey);
+
     const remainingRows = state
         .getIn([stateKey, 'data'])
         .remove(rowIndex || 0, 1);
 
     const record = state.get(stateKey);
 
+    const updatedTotal = existingState
+    && existingState.get('total')
+    ? existingState.get('total')
+    : 0;
+
     const updated = record.merge({
         data: remainingRows,
         proxy: remainingRows,
         currentRecords: remainingRows,
+        total: updatedTotal - 1,
```

Comment:
This is a nit, but if total resolved to 0, then we're setting the total to `-1` which probably isn't best.